iothub/device/src/Transport/Amqp/AmqpConnectionHolder.cs

```diff
@@ -174,11 +144,16 @@ public async Task<AmqpIotConnection> EnsureConnectionAsync(CancellationToken can
 
             AmqpIotConnection amqpIotConnection = null;
             IAmqpAuthenticationRefresher amqpAuthenticationRefresher = null;
+
             try
             {
                 await _lock.WaitAsync(cancellationToken).ConfigureAwait(false);
 
-                if (_amqpIotConnection == null || _amqpIotConnection.IsClosing())
+                if (_amqpIotConnection != null && !_amqpIotConnection.IsClosing())
```

Comment:
Change the order so the smallest action is handled first, to avoid loss of reader context further down.